File: python-algorithm/ds_6_string/L485.cpp
//
// Created by 李恒 on 2020/7/1.
//
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int res = 0;
        int tmp = 0;
        for(int i=0; i<nums.size(); ++i){
            if(nums[i] == 0){
                res = max(res, tmp);
                tmp = 0;
            }
            else
                ++tmp;
        }
        return max(res, tmp); //不是以0结尾的时候最后一次tmp要额外处理
    }
};


class Solution1 {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int res = 0;
        int tmp = 0;
        for(int i=0; i<nums.size(); ++i){
            if(nums[i] == 0)
                tmp = 0;
            else
                ++tmp;
            res = max(res, tmp);
        }
        return res;
    }
};

File: python-algorithm/ds_6_string/L485.py
# -*- coding: utf-8 -*-
# @Time    : 2020/10/3 13:59
# @Author  : No Name


class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        res = 0
        i = 0
        while i<len(nums):
            tmp_len = 0
            while i<len(nums) and nums[i] == 1:
                tmp_len += 1
                i += 1
            if tmp_len != 0:
                res = max(res, tmp_len)
            else:
                i += 1
        return res

    
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        res = 0
        tmp_len = 0
        for n in nums:
            if n == 0:
                res = max(res, tmp_len)
                tmp_len = 0
            else:
                tmp_len += 1
        return max(res, tmp_len) # 不是以0结尾的时候最后一次tmp要额外处理

